dApp Kit Next docs (#529)
MystenLabs/ts-sdks: packages/docs/content/dapp-kit-next/migration.mdx

npm uninstall @mysten/dapp-kit @tanstack/react-query

# Install new packages
npm i @mysten/dapp-kit-react @mysten/dapp-kit-core @mysten/sui
```

### 2. Replace Provider Setup

Replace multiple nested providers with a single unified provider.

**Before:**

```tsx
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { SuiClientProvider, WalletProvider } from '@mysten/dapp-kit';

export function ExampleComponent() {
	const queryClient = new QueryClient();
	const networkConfig = {
		mainnet: { url: 'https://mainnet.sui.io:443' },
		testnet: { url: 'https://testnet.sui.io:443' },
	};

	return (
		<QueryClientProvider client={queryClient}>
			<SuiClientProvider networks={networkConfig} defaultNetwork="mainnet">
				<WalletProvider>
					<App />
				</WalletProvider>
			</SuiClientProvider>
		</QueryClientProvider>
	);
}
```

**After:**

```tsx
import { DAppKitProvider, createDAppKit } from '@mysten/dapp-kit-react';
import { getFullnodeUrl, SuiClient } from '@mysten/sui/client';

export function ExampleComponent() {
	const dAppKit = createDAppKit({
		networks: ['mainnet', 'testnet'],
		defaultNetwork: 'testnet',
		createClient: (network) =>

> 			new SuiClient({

Reviewer: I think we probably want to register the global type as part of the migration too, since the useSuiClient calls won't work as expected without that step

import { SuiClientProvider, WalletProvider } from '@mysten/dapp-kit';

export function App() {
	const queryClient = new QueryClient();
	const networkConfig = {
		mainnet: { url: 'https://mainnet.sui.io:443' },
		testnet: { url: 'https://testnet.sui.io:443' },
	};

	return (
		<QueryClientProvider client={queryClient}>
			<SuiClientProvider networks={networkConfig} defaultNetwork="mainnet">
				<WalletProvider>
					<App />
				</WalletProvider>
			</SuiClientProvider>
		</QueryClientProvider>
	);
}
```

**After:**

```tsx
// App.tsx
import { DAppKitProvider } from '@mysten/dapp-kit-react';
import { dAppKit } from './dapp-kit.ts';

export function App() {
	return (
		<DAppKitProvider dAppKit={dAppKit}>
			<App />
		</DAppKitProvider>
	);
}
```

### 4. Update Hook Usage

Most existing hooks continue to work with no changes needed:

**No changes required:**

- `useCurrentAccount`
- `useSuiClient`
- `useCurrentWallet`

**New hooks available:**

- `useDAppKit` - Access the main dAppKit instance
- `useConnection` - Get current wallet connection
- `useCurrentNetwork` - Get current network

### 5. Replace Mutation Patterns

> The new version uses direct promise-based calls instead of React Query mutations.

Reviewer 2: I think this should mention something about the new version of dApp Kit not integrating as tightly with TanStack Query to give developers flexibility with their preference of tech stack. The `after` example could give an example of how to integrate with tanstack query (i.e. literally just pass the function into `useMutation({mutationFn: signAndExecuteTransaction})`

cc @hayes-mysten

**Before (React Query mutations):**

```tsx
import { useEffect } from 'react';
import { useSignAndExecuteTransaction } from '@mysten/dapp-kit';
import type { Transaction } from '@mysten/sui/transactions';

export function ExampleComponent({ transaction }: { transaction: Transaction }) {
	const { mutateAsync: signAndExecute } = useSignAndExecuteTransaction();

> 	useEffect(() => {

Reviewer 2: I think having this in a `useEffect` and async logic kind of confusing for general n00bs reading the documentation, I think it'd be preferable to have a button where the onClick calls the mutation function

	const dAppKit = useDAppKit();

	useEffect(() => {
		async function execute() {
			try {
				const result = await dAppKit.signAndExecuteTransaction({ transaction });
				console.log(result);
			} catch (error) {
				console.error(error);
			}
		}
		execute();
	}, [dAppKit]);

	// ...
}
```

### 6. Replace Data Fetching Patterns

React Query hooks are no longer available. Use standard React patterns for data fetching.

**Before (React Query hooks):**

```tsx
import { useSuiClientQuery } from '@mysten/dapp-kit';

export function ExampleComponent({ objectId }: { objectId: string }) {
	const { data, isLoading, error } = useSuiClientQuery('getObject', {
		id: objectId,
	});
	// ...
}
```

**After (Standard React patterns):**

```tsx
import { useSuiClient } from '@mysten/dapp-kit-react';
import { useState, useEffect } from 'react';
import type { SuiObjectData } from '@mysten/sui/client';

export function ExampleComponent({ objectId }: { objectId: string }) {
	const client = useSuiClient();
	const [data, setData] = useState<SuiObjectData | null>(null);
	const [isLoading, setIsLoading] = useState(true);
	const [error, setError] = useState<string | null>(null);

> 	useEffect(() => {

Reviewer 2: same re: the react-query before and after

		async function getObject() {
			try {
				const object = await client.getObject({ id: objectId });
				setData(object.data);
			} catch (err: Error | unknown) {
				setError((err as Error)?.message || 'Unknown error');
			} finally {
				setIsLoading(false);
			}
		}
		getObject();
	}, [client, objectId]);
}
```

### 7. Update The Remaining Code

The following hooks from the original dapp-kit are not available in dapp-kit-next:

- `useSuiClientQuery`
- `useSuiClientMutation`
- `useSuiClientInfiniteQuery`
- `useResolveSuiNSNames`

**Migration:** Use `useSuiClient()` directly with standard React patterns or integrate your
preferred data fetching solution.

<Callout type="warning">

> 	The `reportTransactionEffects` feature has been removed from the [Wallet

Reviewer 2: ```suggestion
	The `reportTransactionEffects` feature is planned for deprecation in the [Wallet
```
